logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java

```diff
@@ -37,19 +37,36 @@
     private long maxAgeDays = 30; // GDPR rules: max 30 days
     private long sizeLimit = 30L * (1L << 30); // 30 GB
 
+    private void run() {
+        try {
+            Thread.sleep(125000); // 2 m 5 s
+            while (true) {
+                while (maintenance()) {
```

Comment:
The maintenance thread will stop if `maintenance()` throws an unchecked exception.

Reply:
yes.